source/host/Option.cc

```diff
@@ -0,0 +1,52 @@
+#include "Option.hh"
+
+#include <cstring>
+
+namespace Host::Option {
+
+std::optional<EOption> CheckFlag(const char *arg) {
+    ASSERT(arg);
+    if (arg[0] != '-') {
+        return std::nullopt;
+    }
+
+    // Verbose flag
+    if (arg[1] == '-') {
+        const char *verbose_arg = &arg[2];
+
+        if (strcmp(verbose_arg, "mode") == 0) {
+            return EOption::Mode;
+        }
+
+        if (strcmp(verbose_arg, "suite") == 0) {
+            return EOption::Suite;
+        }
+
+        return EOption::Invalid;
+    } else {
+        switch (arg[1]) {
+        case 'M':
+        case 'm':
+            return EOption::Mode;
+        case 'S':
+        case 's':
+            return EOption::Suite;
+        default:
+            return EOption::Invalid;
+        }
+    }
+
+    // This is unreachable
+    return std::nullopt;
+}
+
+EMode CheckModeArg(const char *arg) {
+    ASSERT(arg);
+    if (strcmp(arg, "test") == 0) {
```

Comment:
This would mean that every additional mode requires its own if-statement. I think a possible alternative (and probably my preference) is if we used a std::unordered_map<std::string, EMode> so we can make the association between args and their enum representation more clearly readable, and the parsing would ultimately just be a table lookup, with `EMode::Invalid` if not found in the table.